### DISCUSSION TITLE
Developer guide for making tools compatible with nushell?

### DISCUSSION THREAD

**Discussion by MilesCranmer** [nushell/nushell#16657] [Q&A]
### Question

I maintain this shell tool rip2: https://github.com/MilesCranmer/rip2. I also use nushell. How can I make `rip` nushell-compatible, so that it outputs normally to bash, but outputs nushell records to nushell?

Would I need to make a proper integration into nushell?

**Comment by fdncred** [comment]
I've seen a couple implementations for this. [One](https://github.com/awdemos/demo-rust-server/tree/master), most recently, just added a dependency on nu-table and wrote it out that way. The other way is to output nuon or json and then use nushell translate from nuon/json.

**Comment by MilesCranmer** [comment]
Thanks. From a design perspective, what's your opinion on if I just add a feature to my package so you could do, e.g.,

```
cargo install --features nu --locked rip2
```

to install with nushell table support. Then I can keep my binaries minimal for non-nushell users.

Wdyt?

**Comment by fdncred** [comment]
I think that's a good way to keep the minimum number of dependencies. On the other hand, if you want to switch functionality in real time, a parameter may be better.

**Comment by WindSoilder** [comment]
Closing the issue because the question is answered.

**Comment by MilesCranmer** [comment]
@WindSoilder would be nice to keep open until there's docs or a developer guide for doing this

**Comment by WindSoilder** [comment]
I'm afraid is there going to be any guide for doing this.  Because different tools may use some of nushell crates in different way.
But yeah, let's reopen this if you think it's necessary.
